kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/BatchesResource.scala

```diff
@@ -561,6 +561,58 @@ private[v1] class BatchesResource extends ApiRequestContext with Logging {
     }
   }
 
+  @ApiResponse(
+    responseCode = "200",
+    content = Array(new Content(
+      mediaType = MediaType.APPLICATION_JSON,
+      schema = new Schema(implementation = classOf[ReassignBatchResponse]))),
+    description = "Reassign instance's all batch sessions to a new instance")
+  @GET
+  @Path("{kyuubiInstance}/{batchId}/reassign")
+  def reassignBatchSessions(
+      @PathParam("kyuubiInstance") kyuubiInstance: String,
+      @PathParam("batchId") batchId: String,
+      @QueryParam("force")
+      @DefaultValue("false") force: Boolean): ReassignBatchResponse = {
+    sessionManager.getBatchMetadata(batchId).map { metadata =>
+      val batchInstance = metadata.kyuubiInstance
+      if (batchInstance != kyuubiInstance) {
+        throw new IllegalStateException(s":parameter $kyuubiInstance needs to match $batchInstance")
+      }
+      val url = fe.connectionUrl
+      if (kyuubiInstance == url) {
+        throw new IllegalStateException(s"KyuubiInstance is alive: $kyuubiInstance")
+      }
+      if (!force && checkInstanceNetwork(kyuubiInstance, batchId)) {
+        throw new IllegalStateException(s"KyuubiInstance is alive: $kyuubiInstance," +
+          s"please use force option to bypass this check")
+      }
+      try {
+        sessionManager.reassignBatchSessions(kyuubiInstance, fe.connectionUrl)
```

Comment:
the logical is not correct,

you check the `batch` exists and then try to reassign all the batches on the instance. 

It is not align.